### PR TITLE
content(js): add variable article

### PR DESCRIPTION
**Pull request by ddivad195**
Add a new article for Variables in Javascript.

Added some basic information and usage about JS Variables, discussed dynamic typing & discussed let and const.

### CODE REVIEW

**Comment by ddivad195** [comment]
Hey, so I was going to also include a section talking about `var` and why it should not be used anymore. What are your thoughts on this @veksen @Xetera ? Should I add that here, or does it belong in another article? Not sure if it would just add confusion here or not.

**Comment by Xetera** [comment]
I think it would belong here, yeah

**Comment by veksen** [comment]
Another thing I think belongs in this article is the undeclared variables landing in global scope and maybe an example of why they're bad. It's a pretty common mistake for beginners, especially for those coming from something like PHP.

**Comment by ddivad195** [comment]
Added a section about `var` and the problems that using it can have. Let me know if I need to make any other changes!

File: src/content/resources/javascript/variables.md [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+There are a number of problems that come from using `var` to declare variables, that are fixed by using `let` or `const` instead. 
+
+### Block Scoping
+Var is not "block scoped" (a block is anything that contains `{}`, so functions, if, for loop, etc...) This means that variables declared using `var` are not *just* defined in the block they are declared, but declared and can be accessible globally. This can have un-intended side effects.
+
+```js
+function myFunction() {
+  var myVar = "test" //declared with the 'var' keyword
+}
+console.log(myVar); // variable can be used outside myFunction
```
Comment (Aiden01) [inline src/content/resources/javascript/variables.md]: Little error here, ``var`` is function-scoped (or global if it's not declared inside a function), so ``myVar`` isn't accessible outside of ``myFunction``

**Comment by veksen** [comment]
I rebased and force pushed the branch on your repo @ddivad195

File: src/content/resources/javascript/variables.md (hunk 1 above)
Comment (ddivad195) [reply src/content/resources/javascript/variables.md]: Updated now, thanks for the spotting that :)

**Comment by veksen** [comment]
From a quick proof read looks good to me, could you run `npm run format` to keep it consistent? (It also formats the code examples in MD which is nice!)

**Comment by ddivad195** [comment]
Done @veksen , thanks!

**Pull request merged**